server/utils.js

```diff
@@ -28,17 +34,26 @@ var convertLatLngToStreet = function(lng, lat, callback) {
 
 var convertDirectionsToStreet = function(req, callback) {
   var responseObj = req.body.streets;
-  for (var i = 0; i < req.body.length; i++) {
-    utils.convertLatLngToStreet(req.body.streets[i][0][0], req.body.streets[i][0][1], function(err, results) {
-      if (err) {
-        callback(err, null);
-      } else {
-        responseObj[i].push(results);
-      }
-    });
-    if (i === req.body.length - 1) {
-      callback(responseObj);
-    }
+  var counter = 0;
+  for (var i = 0; i < req.body.streets.length; i++) {
+    (function(i){
+      setTimeout(function() {
+        convertLatLngToStreet(req.body.streets[i][0][0], req.body.streets[i][0][1], function(err, results) {
+          if (err) {
+            callback(err, null);
```

Comment:
im honestly not sure how to best handle this error, its tricky b/c its in a for loop. i am making some refactors and we can see how it looks then